pkg/k8s/resource_helper.go

```diff
@@ -73,9 +74,20 @@ func (rh *ResourceHelper) NewResourcesFromFilename(filename string) ([]*Resource
 	decoder := yaml.NewYAMLReader(reader)
 
 	for {
-		bytes, err := decoder.Read()
+		bs, err := decoder.Read()
 
-		if len(bytes) == 0 {
+		if err != nil {
+			return []*Resource{}, fmt.Errorf("decode doc from %s: %s", filename, err.Error())
+		}
+
+		// Converting to JSON and looking for "null" ignores empty docs.
+		bs, err = yaml.ToJSON(bs)
+
+		if err != nil {
+			return []*Resource{}, fmt.Errorf("failed to convert yaml to json %s: %s", filename, err.Error())
+		}
+
+		if len(bs) == 0 || bytes.Equal(bs, []byte("null")) {
```

Comment:
I've just noticed a bug here - we return from the function and quit parsing the manifest if we get a `null` document, which means that any manifest which starts with an empty document will have all its resources skipped over